erizo/src/erizo/MediaStream.cpp

```diff
@@ -0,0 +1,515 @@
+
+/*
+ * MediaStream.cpp
+ */
+
+#include <cstdio>
+#include <map>
+#include <algorithm>
+#include <string>
+#include <cstring>
+#include <vector>
+
+#include "./MediaStream.h"
+#include "./SdpInfo.h"
+#include"./WebRtcConnection.h"
+#include "rtp/RtpHeaders.h"
+#include "rtp/RtpVP8Parser.h"
+#include "rtp/RtcpAggregator.h"
+#include "rtp/RtcpForwarder.h"
+#include "rtp/RtpSlideShowHandler.h"
+#include "rtp/RtpTrackMuteHandler.h"
+#include "rtp/BandwidthEstimationHandler.h"
+#include "rtp/FecReceiverHandler.h"
+#include "rtp/RtcpProcessorHandler.h"
+#include "rtp/RtpRetransmissionHandler.h"
+#include "rtp/RtcpFeedbackGenerationHandler.h"
+#include "rtp/RtpPaddingRemovalHandler.h"
+#include "rtp/StatsHandler.h"
+#include "rtp/SRPacketHandler.h"
+#include "rtp/SenderBandwidthEstimationHandler.h"
+#include "rtp/LayerDetectorHandler.h"
+#include "rtp/LayerBitrateCalculationHandler.h"
+#include "rtp/QualityFilterHandler.h"
+#include "rtp/QualityManager.h"
+#include "rtp/PliPacerHandler.h"
+#include "rtp/RtpPaddingGeneratorHandler.h"
+#include "rtp/RtpUtils.h"
+
+namespace erizo {
+DEFINE_LOGGER(MediaStream, "MediaStream");
+
+MediaStream::MediaStream(
+  std::shared_ptr<WebRtcConnection> connection,
+  const std::string& media_stream_id) :
+    audioEnabled_{false}, videoEnabled_{false},
+    connection_{connection},
+    stream_id_{media_stream_id},
+    bundle_{false},
+    pipeline_{Pipeline::create()}, audio_muted_{false}, video_muted_{false},
+    pipeline_initialized_{false} {
+  setVideoSinkSSRC(kDefaultVideoSinkSSRC);
+  setAudioSinkSSRC(kDefaultAudioSinkSSRC);
+  ELOG_INFO("%s message: constructor, id: %s",
+      toLog(), media_stream_id.c_str());
+  source_fb_sink_ = this;
+  sink_fb_source_ = this;
+  stats_ = connection->getStatsService();
+  quality_manager_ = std::make_shared<QualityManager>();
+  packet_buffer_ = std::make_shared<PacketBufferService>();
+  worker_ = connection->getWorker();
+
+  rtcp_processor_ = std::make_shared<RtcpForwarder>(static_cast<MediaSink*>(this), static_cast<MediaSource*>(this));
+
+  shouldSendFeedback_ = true;
```

Comment:
we might take the opportunity to convert these var names to snake_case (just in case you think it worths changing it now)